### ISSUE TITLE
iosxe_system attributes are not removed

### ISSUE THREAD

**Issue by grotewortel** [CiscoDevNet/terraform-provider-iosxe#48]
Hi,

The settings from `iosxe_system` do not seem to be removed from the Cisco device after removing them from the terraform config.
I am using Cisco IOS XE Software, Version 17.11.01a and iosxe provider 0.4.0.

The following takes you through a test scenario:

**Starting terraform config:**

```
terraform {
    required_providers {
        iosxe = {
        source  = "CiscoDevNet/iosxe"
        }
    }
}

provider "iosxe" {
    alias = "router8001"
    insecure = true
    username = "username"
    password = "password"
    url      = "https://1.2.3.4"
}

resource "iosxe_system" "example" {
  provider = iosxe.router8001
  hostname             = "ROUTER8001"
  ipv6_unicast_routing = true
  ip_source_route      = false
  ip_domain_lookup     = false
  ip_domain_name       = "test.com"
  login_delay          = 1
  login_on_failure     = true
  login_on_failure_log = true
  login_on_success     = true
  login_on_success_log = true
  multicast_routing_vrfs = []
}
```

**Config after apply:**
```
ROUTER8001#show run | i domain name
ip domain name test.com
ROUTER8001#show run | i delay      
login delay 1
ROUTER8001#
```

**Modified terraform config:**
`ip_domain_name` and `login_delay` have been commented out
```
resource "iosxe_system" "example" {
  provider = iosxe.router8001
  hostname             = "ROUTER8001"
  ipv6_unicast_routing = true
  ip_source_route      = false
  ip_domain_lookup     = false
  # ip_domain_name       = "test.com"
  # login_delay          = 1
  login_on_failure     = true
  login_on_failure_log = true
  login_on_success     = true
  login_on_success_log = true
  multicast_routing_vrfs = []
}
```

**Apply output of the modified terraform config:**
Both changes are detected and are promised to be applied.
```
jan@Jans-MBP cisco % terraform apply -auto-approve
iosxe_system.example: Refreshing state... [id=Cisco-IOS-XE-native:native]

Terraform used the selected providers to generate the following execution plan. Resource actions are
indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # iosxe_system.example will be updated in-place
  ~ resource "iosxe_system" "example" {
        id                     = "Cisco-IOS-XE-native:native"
      - ip_domain_name         = "test.com" -> null
      - login_delay            = 1 -> null
        # (9 unchanged attributes hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
iosxe_system.example: Modifying... [id=Cisco-IOS-XE-native:native]
iosxe_system.example: Modifications complete after 0s [id=Cisco-IOS-XE-native:native]

Apply complete! Resources: 0 added, 1 changed, 0 destroyed.
```

**Config after apply:**
Both options are still present.
```
ROUTER8001#show run | i domain name
ip domain name test.com
ROUTER8001#show run | i delay 
login delay 1
```

Please advise

Jan

**Comment by grotewortel** [comment]
In Terraform the `iosxe_system` resource is still managed and the restconf data also seems to have a `login` datastructure (where `delay` indeed is removed.

**with login delay set**
```{
 "data": {
  "Cisco-IOS-XE-native:native": {
    ...
    "login": {
      "delay": 1,
      "on-failure": {
        "log": {
        }
      },
      "on-success": {
        "log": {
        }
      }
    },
    ...
  }
 }
}
```


**after conf t / no login delay**
```
 "data": {
  "Cisco-IOS-XE-native:native": {
    ...
    "login": {
      "on-failure": {
        "log": {
        }
      },
      "on-success": {
        "log": {
        }
      }
    },
    ...
  }
 }
}
```

**Comment by grotewortel** [comment]
I added attributes to the subject for more clarity.

**Comment by danischm** [comment]
The json outputs you added above, are those meant to confirm what I tried to explain or are you saying that it is not working as expected?

**Comment by grotewortel** [comment]
What I am trying to say is that it does not look like this is a resource that is no longer managed by terraform. On the terraform side there is a resource `iosxe_system` that is still present. So it is still managed. The apply after removing the attribute from the resource shows that action is taken (`- login_delay = 1 -> null`). 

The json shows that in both situations there is an object `login` present. I would expect this `login` object to be considered managed if terraform has an `iosxe_system` resource.

My perception is that, in general, if you remove a resource from a terraform configuration, is does not mean that it becomes unmanaged. The behaviour I would expect is that the resource is being deleted.
For instance, if you deploy an VM/EC2 instance on AWS via terraform and you remove that VM from the configuration, the VM will be terminated.

PS: I am very happy with the swift response. Please consider this positive and constructive feedback :-)

**Comment by danischm** [comment]
https://github.com/CiscoDevNet/terraform-provider-iosxe/commit/21321a65b299573628c62be252e88ec83e6dfab6

**Comment by grotewortel** [comment]
Nice! Do you expect to release 0.4.1 soon?

**Comment by danischm** [comment]
Should be available within a week.

**Comment by grotewortel** [comment]
I confirm that it is fixed in 0.5.0. 
Thank you!
